src/liquid/components/ld-header/ld-header.tsx

```diff
@@ -0,0 +1,136 @@
+import { Component, Host, h, Prop, Element, Watch } from '@stencil/core'
+import { getClassNames } from 'src/liquid/utils/getClassNames'
+
+/**
+ * @virtualProp ref - reference to component
+ * @virtualProp {string | number} key - for tracking the node's identity when working with lists
+ * @part container - Actual header element that limits the width of the header content
+ * @part logo - `ld-icon` element containing the default logo
+ * @part site-name - `ld-typo` element containing the site name
+ */
+@Component({
+  assetsDirs: ['assets'],
+  tag: 'ld-header',
+  styleUrl: 'ld-header.css',
+  shadow: true,
+})
+export class LdHeader {
+  @Element() el: HTMLElement
+  private lastOffset?: number
+  private currentHeight?: number
+
+  /** Hides header. */
+  @Prop({ mutable: true }) hidden = false
+
+  /** Hide the header when the user scrolls down and show it again, when the user scrolls up. */
+  @Prop() hideOnScroll = false
+
+  /** Title attribute of the logo link. */
+  @Prop() logoTitle?: string
+
+  /** URL that the logo links to. */
+  @Prop() logoUrl?: string
+
+  /** Make the header sticky. */
+  @Prop() sticky = false
+
+  /** Name shown on the right side of the logo. */
+  @Prop() siteName?: string
+
+  private updateScrollDirection = () => {
+    const offset = window.pageYOffset ?? document.documentElement.scrollTop
+
+    if (window.innerHeight + offset >= document.body.offsetHeight) {
+      this.hidden = false
+    } else if (offset > this.lastOffset && offset > this.currentHeight) {
+      this.hidden = true
+    } else {
+      this.hidden = false
+    }
+
+    // For mobile or negative scrolling
+    this.lastOffset = offset < 0 ? 0 : offset
+  }
+
+  @Watch('hideOnScroll')
+  connectedCallback() {
+    if (this.hideOnScroll) {
+      this.lastOffset = window.pageYOffset || document.documentElement.scrollTop
+      window.addEventListener('scroll', this.updateScrollDirection, false)
```

Comment:
```suggestion
      window.addEventListener('scroll', this.updateScrollDirection, { passive: true })
```

Reply:
Why did you use the `useCapture` option here in the first place?

Reply:
Changed.